sphinx/application.py

```diff
@@ -1181,6 +1182,13 @@ def add_env_collector(self, collector: Type[EnvironmentCollector]) -> None:
         logger.debug('[app] adding environment collector: %r', collector)
         collector().enable(self)
 
+    def add_html_assets_in_all_pages(self):
+        """Tell extensions to insert HTML assets in all the pages.
+
+        .. versionadded: 4.1
+        """
+        self.html_assets_in_all_pages = True
```

Comment:
Could you store this into the `sphinx.register.SphinxComponentRegistry` instead? It would be better not to have settings or components into the app object directly.

Reply:
Thanks for pointing at this to me. I wasn't aware of the `SphinxComponentRegistry` 😅  I'll move this there once we have defined the name for the value/method.

Reply:
I wish https://github.com/sphinx-doc/sphinx/pull/9215 will help you.

Reply:
Thanks! That PR helped me to understand what's required here and where to store this value. I think I know what to do now 😄 

I'm just waiting for a better name suggestion to update the PR.